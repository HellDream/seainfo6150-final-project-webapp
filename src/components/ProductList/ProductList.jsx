import React from 'react';
import ProductItem from '../ProductItem/ProductItem';
import styles from './ProductList.module.css';

const ProductList = (props) => {
    const products = props.products;
    return (
        <ul className={styles.productContainer}>
            {products.map((product) => (
                <ProductItem  product={product} key={product.slug} category={props.categoryTitle}/>
            ))}
        </ul>
    );
};
export default ProductList;
