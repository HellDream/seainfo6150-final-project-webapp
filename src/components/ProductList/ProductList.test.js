import React from 'react';
import ProductList from './ProductList';
import { render } from '@testing-library/react'

import img from '../../images/books.jpg';
const products = [{
    slug: 1,
    imageUrl: img,
    title: 'book',
    price:2.33,

}];
describe('ProductList tests', () => {
    it('renders correctly', () => {
        const { container } = render(<ProductList products={products} categoryTitle="testTitle"/>);
        expect(container).toMatchSnapshot();
    });
});
  