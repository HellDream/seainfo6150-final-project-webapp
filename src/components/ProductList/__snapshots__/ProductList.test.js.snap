// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList tests renders correctly 1`] = `
<div>
  <ul
    class="productContainer"
  >
    <li
      class="product"
    >
      <a
        class="link"
        href="/testTitle/product/1"
      >
        <div
          class="imageBlock"
        >
          <img
            alt="book"
            class="image"
            src="books.jpg"
          />
        </div>
        <div
          class="productInfoBlock"
        >
          <h3
            class="title"
          >
            book
          </h3>
          <h4
            class="price"
          >
            $
            2.33
          </h4>
        </div>
        <time
          class="time"
        >
          a few seconds ago
        </time>
      </a>
    </li>
  </ul>
</div>
`;
